gateway/src/main/java/io/camunda/zeebe/gateway/impl/stream/ClientStreamAdapter.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.gateway.impl.stream;
+
+import static io.camunda.zeebe.util.buffer.BufferUtil.wrapString;
+
+import io.camunda.zeebe.gateway.ResponseMapper;
+import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.ActivatedJob;
+import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.StreamActivatedJobsRequest;
+import io.camunda.zeebe.msgpack.value.StringValue;
+import io.camunda.zeebe.protocol.impl.stream.job.ActivatedJobImpl;
+import io.camunda.zeebe.protocol.impl.stream.job.JobActivationProperties;
+import io.camunda.zeebe.protocol.impl.stream.job.JobActivationPropertiesImpl;
+import io.camunda.zeebe.scheduler.future.ActorFuture;
+import io.camunda.zeebe.transport.stream.api.ClientStreamConsumer;
+import io.camunda.zeebe.transport.stream.api.ClientStreamId;
+import io.camunda.zeebe.transport.stream.api.ClientStreamer;
+import io.grpc.stub.ServerCallStreamObserver;
+import java.util.concurrent.CompletableFuture;
+import org.agrona.DirectBuffer;
+
+public class ClientStreamAdapter {
+  private final ClientStreamer<JobActivationProperties> jobStreamer;
+
+  public ClientStreamAdapter(final ClientStreamer<JobActivationProperties> jobStreamer) {
+    this.jobStreamer = jobStreamer;
+  }
+
+  public void handle(
+      final StreamActivatedJobsRequest request,
+      final ServerCallStreamObserver<ActivatedJob> responseObserver) {
+    final JobActivationProperties jobActivationProperties = toJobActivationProperties(request);
+
+    final ActorFuture<ClientStreamId> clientStreamId =
+        jobStreamer.add(
+            wrapString(request.getType()),
+            jobActivationProperties,
+            new ClientStreamConsumerImpl(responseObserver));
+
+    final var removeJobStream = new JobStreamRemover(clientStreamId, jobStreamer);
+    responseObserver.setOnCloseHandler(removeJobStream);
+    responseObserver.setOnCancelHandler(removeJobStream);
+  }
+
+  private JobActivationProperties toJobActivationProperties(
+      final StreamActivatedJobsRequest request) {
+    final JobActivationPropertiesImpl jobActivationProperties = new JobActivationPropertiesImpl();
+    final DirectBuffer worker = wrapString(request.getWorker());
+    jobActivationProperties.setWorker(worker, 0, worker.capacity());
+    jobActivationProperties.setTimeout(request.getTimeout());
+    jobActivationProperties.setFetchVariables(
+        request.getFetchVariableList().stream().map(StringValue::new).toList());
```

Comment:
:wrench: I'm wondering if we should set the variables simialrly to [here](https://github.com/camunda/zeebe/blob/main/gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/request/BrokerActivateJobsRequest.java#L45-L47).

Just to make sure that we avoid the bug we encountered in the [`JobActivationPropertiesImpl`](https://github.com/camunda/zeebe/blob/main/protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/stream/job/JobActivationPropertiesImpl.java#L54-L58).

Reply:
The issue with [JobActivationPropertiesImpl](https://github.com/camunda/zeebe/pull/13323#discussion_r1255916587) was happening while iterating over `ArrayProperty` which internally uses `ArrayValue`'s iterator. Then, `ArrayValue`'s iterator uses only one buffer object that contains the strings. Because of that, we ended up with one buffer reference.

This case is independent from `ArrayValue` implementation and creates a new `StringValue` by iterating over a list, so should be a different one.

What do you think? @koevskinikola

Reply:
Yup, all good. We can leave it as it is :+1: